app/src/main/java/org/sil/storyproducer/tools/Utilities.kt

```diff
@@ -113,3 +115,20 @@ fun getFreeExternalMemorySize(): Long {
     }
     return 0
 }
+
+
+fun getMaxFreeExternalMemoryFile(): File? {
+    var maxFree = 0L;
+    var maxFreeDir: File? = null
+    val externalFilesDirs = ContextCompat.getExternalFilesDirs(App.appContext, null)
```

Comment:
If a shared storage device is emulated (as determined by [Environment#isExternalStorageEmulated(File)](https://developer.android.com/reference/android/os/Environment#isExternalStorageEmulated(java.io.File))), it's contents are backed by a private user data partition

Reply:
So, if an external storage files folder is emulated there is no point in using it to save internal storage.